src/components/PeopleFilters.tsx

```diff
@@ -1,12 +1,55 @@
+import classNames from 'classnames';
+import { useSearchParams } from 'react-router-dom';
+import { SearchLink } from './SearchLink';
+import { FilterParam } from '../types/FilterParams';
+import { Sex } from '../types/Sex';
+import { getSearchWith } from '../utils/searchHelper';
+
 export const PeopleFilters = () => {
+  const CENTURIES = ['16', '17', '18', '19', '20'];
+
+  const [searchParams, setSearchParams] = useSearchParams();
+  const query = searchParams.get(FilterParam.Query) || '';
+  const selectedCenturies = searchParams.getAll(FilterParam.Centuries) || [];
+  const sex = searchParams.get(FilterParam.Sex) || '';
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchParams(getSearchWith(searchParams, {
+      [FilterParam.Query]: event.target.value || null,
+    }));
+  };
+
+  const handleCenturiesChange = (century: string) => {
+    return (
+      selectedCenturies.includes(century)
+        ? selectedCenturies.filter(year => year !== century)
+        : [...selectedCenturies, century]
+    );
+  };
```

Comment:
It is better to define this function outside the component so that it is not created every time a new render is performed.